src/pages/Portfolio/MyKiva/MyKivaPage.vue

```diff
@@ -161,6 +168,21 @@ const handleBadgeClicked = badge => {
 	showBadgeModal.value = true;
 };
 
+const handleBadgeLevelClicked = level => {
+	tier.value = level;
+	const selectedTier = badgeData?.achievementData?.tiers?.find(t => t.level === level);
+	state.value = selectedTier?.completedDate ? STATE_EARNED : STATE_IN_PROGRESS;
+};
+
+const handleBadgeModalClosed = () => {
+	if (state.value === STATE_JOURNEY) {
+		showBadgeModal.value = false;
+		return;
+	}
+
+	state.value = STATE_JOURNEY;
```

Comment:
Do we want to return to journey here or close the modal? It isn't really clear in the Figma.

Reply:
Sounds good for first pass.

Reply:
This should return to the journey